### DISCUSSION TITLE
Feature Request: Auto-Tip

### DISCUSSION THREAD

**Discussion by ghost** [stackernews/stacker.news#1874] [Ideas]
There is still too much friction associated with making Lightning payments on Stacker News.

Right now, every user must click the small Lightning bolt next to a post or comment to pay that creator.

The result is that across the site, many users will respond to comments and posts and engage in thoughtful dialogue without ever spending money to show their appreciation for the content they're replying to.

To make it easier for users to spend sats on great content, Stacker News should enable an opt-in "auto-tip" feature that automatically tips a user's default tip amount on every post or comment they reply to. This way, the user only needs to reply to a comment or post and no longer needs the extra Lightning bolt click (although the Lightning bolt will still be there if they want to tip more).

This feature could also be expanded to automatically tip every post that a user clicks on from the main page.

**Comment by cointastical** [comment]
Elegant opt-in solution.  I like it. 

The only suggestion I might have is where I have flagged a post and then go to comment (e.g., to say why I flagged it), I would not then want the tip to automatically occur.  So the logic on this would be to check first if I flagged to post, then to not trigger the auto-tip when I do a reply.

**Comment by huumn** [comment]
It might be worth having something more generic. I'd say there's 10% of occasions where I comment and don't want to also tip (nor do I want to flag). 

Perhaps if you enable auto-tipping, the ui of commenting changes slightly and there's an auto-tip checkbox next to the submit button.

**Comment by huumn** [comment]
Related #758

**Comment by adlai** [comment]
there are several different situations where you feel gratitude for the site providing you some item, rather than annoyance at the delay of interesting content:

1. someone answered your question; these aren't necessarily `A:"..."`/`B:"...?"`/`A"!"` reply chains, or `A:"@B"`/`B:"..."` chains, because you might have asked a general question and someone provided the answer unsolicited
2. someone already posted a reply saying pretty much what you would have, saving you the trouble of producing the wording, let alone the cost of posting it
3. media that wasn't necessarily informative or interesting, although entertained you
4. some comment wasn't any of the above and yet restored your faith in humanity [e.g. someone setting a good example by remaining courteous in a situation where nobody else was being polite]

obviously there are also random zaps, and sometimes conversations aren't utilitarian enough for my sterilised github imagination to figure out why they not only happened but kept people in SN rather than some unmonetized or even antimonetary site.

my purpose in enumerating the above scenarios is to help spec out the different behaviors that might be reasonable for an autozap; e.g., a sanely minimal autozap would credit a media uploader the cost of your commenting "LOL"

the sort of crazy idea I often have is that a zap amount could be computed from things like the [mixing entropy](https://en.wikipedia.org/wiki/Entropy_of_mixing#Relationship_to_information_theory) of the comment chain, or maybe the differential increase in the mixing entropy that was added by your comment.
